BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,41 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
-
-        var concatenation = Base58.Bitcoin.Decode(stringValue.AsSpan(3)).ToArray();
-        var hashedPublicKey = concatenation[..20];
-        var givenChecksum = concatenation[20..];
+        const string pattern = @"^did\:e\:(.+?)\:dids\:(.+)(.{2})$";
+        try
+        {
+            var matches = Regex.Matches(stringValue, pattern, RegexOptions.IgnoreCase).First().Groups;
 
-        var realmBytes = Encoding.UTF8.GetBytes(realm);
-        var correctChecksum = CalculateChecksum(realmBytes, hashedPublicKey);
+            var givenChecksum = matches[3].Value;
+            var calculatedChecksum = CalculateChecksum(stringValue[..^2]);
```

Comment:
If `stringValue` is less than 2 characters long, `stringValue[..^2]` will throw an `ArgumentOutOfRangeException`.

In general, this `VIsValid` function barely does any validation.

Reply:
Well, currently it does, because you never check if the regex validated successfully.
But you're right, as soon as you do this correctly, it will not be necessary to check for the array length.